exporters/trace/jaeger/agent.go

```diff
@@ -32,41 +34,76 @@ type agentClientUDP struct {
 	gen.Agent
 	io.Closer
 
-	connUDP       *net.UDPConn
+	connUDP       udpConn
 	client        *gen.AgentClient
 	maxPacketSize int                   // max size of datagram in bytes
 	thriftBuffer  *thrift.TMemoryBuffer // buffer used to calculate byte size of a span
 }
 
+type udpConn interface {
+	Write([]byte) (int, error)
+	SetWriteBuffer(int) error
+	Close() error
+}
+
+type agentClientUDPParams struct {
+	HostPort                   string
+	MaxPacketSize              int
+	Logger                     *log.Logger
+	DisableAttemptReconnecting bool
```

Comment:
```suggestion
	AttemptReconnecting bool
```

Stating this in the affirmative and negating it where necessary maintains consistency with the `AttemptReconnectInterval` field and avoids double negation that can require excess cognitive load when interpreting statements referring to this field.